mapswipe_workers/mapswipe_workers/generate_stats/project_stats.py

```diff
@@ -0,0 +1,345 @@
+    # rename columns, ogr2ogr will fail otherwise
+    results_by_task_id_df.rename(columns={
+        0: '0_count',
+        1: '1_count',
+        2: '2_count',
+        3: '3_count'
+
+    }, inplace=True)
+
+    logger.info(f'aggregated results by task id and computed agreement')
+
+    agg_results_df = results_by_task_id_df.merge(tasks_df[['geom', 'task_id']], left_on='task_id', right_on='task_id')
+    logger.info(f'added geometry to aggregated results')
+
+    return agg_results_df
+
+
+def get_progress_by_date(results_df, groups_df):
+    '''
+    for each project we retrospectively generate the following attributes for a given date utilizing the results:
+    number_of_results, cum_number_of_results, progress, cum_progress
+    '''
+
+
+    groups_df['required_results'] = groups_df['number_of_tasks'] * groups_df['number_of_users_required']
+    required_results = groups_df['required_results'].sum()
+    logger.info(f'calcuated required results: {required_results}')
+
+    results_with_groups_df = results_df.merge(groups_df, left_on='group_id', right_on='group_id')
+    results_by_group_id_df = results_with_groups_df.groupby(['project_id_x', 'group_id', 'day']).agg(
+        number_of_tasks=pd.NamedAgg(column='number_of_tasks', aggfunc='min'),
+        number_of_users_required=pd.NamedAgg(column='number_of_users_required', aggfunc='min'),
+        number_of_users=pd.NamedAgg(column='user_id', aggfunc=pd.Series.nunique)
+    )
+    results_by_group_id_df['number_of_results'] = results_by_group_id_df['number_of_users'] * results_by_group_id_df['number_of_tasks']
+    results_by_group_id_df['cum_number_of_users'] = results_by_group_id_df['number_of_users'].groupby(['project_id_x', 'group_id']).cumsum()
+    results_by_group_id_df['number_of_results_progress'] = results_by_group_id_df.apply(lambda row: calc_results_progress(
+        row['number_of_users'],
+        row['number_of_users_required'],
+        row['cum_number_of_users'],
+        row['number_of_tasks'],
+        row['number_of_results']
+    ), axis=1)
+
+    progress_by_date_df = results_by_group_id_df.reset_index().groupby(['day']).agg(
+        number_of_results=pd.NamedAgg(column='number_of_results', aggfunc='sum'),
+        number_of_results_progress=pd.NamedAgg(column='number_of_results_progress', aggfunc='sum')
+    )
+    progress_by_date_df['cum_number_of_results'] = progress_by_date_df['number_of_results'].cumsum()
+    progress_by_date_df['cum_number_of_results_progress'] = progress_by_date_df['number_of_results_progress'].cumsum()
+    progress_by_date_df['progress'] = progress_by_date_df['number_of_results_progress'] / required_results
+    progress_by_date_df['cum_progress'] = progress_by_date_df['cum_number_of_results_progress'] / required_results
+
+    logger.info('calculated progress by date')
+    return progress_by_date_df
+
+
+def get_new_user(day, first_day):
```

Comment:
I'm probably splitting hairs here, but I would rename this to `is_new_user` because `get_new_user` makes me think the function would return a user, not a boolean.